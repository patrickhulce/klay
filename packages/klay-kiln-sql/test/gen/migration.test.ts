const createMigration = require('../../lib/gen/migration').createNewMigrationFile
const utils = require('../utils')

describe('lib/gen/migration.ts', () => {
  describe('#createNewMigrationFile', () => {
    it('should create a fresh migration file', async () => {
      const {kiln} = utils.setup()
      const migration = await createMigration(kiln)
      expect(migration).toMatchSnapshot()
    })

    it('should create an incremental migration file', async () => {
      const {kiln} = utils.setup()
      const queryFn = jest.fn()
      queryFn.mockReturnValueOnce([[{Table: 'users'}]])
      queryFn.mockReturnValueOnce([[{Field: 'age'}], [{Field: 'email'}]])
      queryFn.mockReturnValueOnce([[{Key_name: 'users_unique_email'}]])

      const connection = {query: queryFn}
      const migration = await createMigration(kiln, connection)
      expect(migration).toMatchSnapshot()
    })
  })
})
