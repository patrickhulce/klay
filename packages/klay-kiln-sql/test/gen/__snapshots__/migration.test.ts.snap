// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/gen/migration.ts #createNewMigrationFile should create a migration file 1`] = `
"module.exports = {
  up: async (queryInterface, Sequelize) => {
    await queryInterface.createTable('users', {
        id: {
          primaryKey: true,
          allowNull: false,
          autoIncrement: true,
          type: Sequelize.BIGINT
        },
        age: Sequelize.BIGINT,
        isAdmin: Sequelize.BOOLEAN,
        email: Sequelize.STRING(250),
        password: Sequelize.STRING(32),
        firstName: Sequelize.STRING(100),
        lastName: Sequelize.STRING(100),
        createdAt: Sequelize.DATE(6),
        updatedAt: Sequelize.DATE(6)
      })
    await queryInterface.addIndex('users', {'name':'users_unique_email','unique':true,'fields':['email']})
    await queryInterface.addIndex('users', {'name':'users_unique_firstname_lastname','unique':true,'fields':['firstName','lastName']})
    await queryInterface.addIndex('users', {'name':'users_email_asc__password_asc','fields':[{'attribute':'email','order':'ASC'},{'attribute':'password','order':'ASC'}]})
    await queryInterface.createTable('photos', {
        id: {
          primaryKey: true,
          allowNull: false,
          type: Sequelize.UUID
        },
        ownerId: {
          references: {
            key: 'id',
            model: 'users'
          },
          type: Sequelize.BIGINT
        },
        aspectRatio: Sequelize.DOUBLE,
        metadata: Sequelize.TEXT,
        createdAt: Sequelize.DATE(6),
        updatedAt: Sequelize.DATE(6)
      })
  },
  down: async (queryInterface, Sequelize) => {
    await queryInterface.dropTable('photos')
    await queryInterface.dropTable('users')
  },
}"
`;
